### PR TITLE
OBG 947 balances added

### PR DESCRIPTION
**Pull request by electronicpeter**

### CODE REVIEW

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java [hunk 1]
```diff
@@ -70,7 +78,34 @@ public PaymentInfoBody extractPaymentInfoBody(ProcessResponse result) {
...
         )
         @Mapping(source = "accountNumber", target = "bban")
+        @Mapping(source = "balances", target = "balances", qualifiedByName = "hbciBalancesToProtocolBalances")
         AccountListDetailBody map(BankAccount account);
+
+
+        @Named("hbciBalancesToProtocolBalances")
+        static List<Balance> hbciBalancesToProtocolBalances(BalancesReport hbciBalances) {
+            List<Balance> protocolBalances = new ArrayList<>();
+            if (hbciBalances != null) {
```
Comment (valb3r) [inline opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java:88]: You can immediately return protocolBalances here

```
if (hbciBalances == null) {
	return protocolBalances;
}

add(protocolBalances, "available", hbciBalances.getAvailableBalance());
add(protocolBalances, "credit", hbciBalances.getCreditBalance());
add(protocolBalances, "ready", hbciBalances.getReadyBalance());
add(protocolBalances, "unready", hbciBalances.getUnreadyBalance());
add(protocolBalances, "used", hbciBalances.getUsedBalance());

return protocolBalances;
```

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java [hunk 2]
```diff
@@ -70,7 +78,34 @@ public PaymentInfoBody extractPaymentInfoBody(ProcessResponse result) {
...
+                add(protocolBalances, "credit", hbciBalances.getCreditBalance());
+                add(protocolBalances, "ready", hbciBalances.getReadyBalance());
+                add(protocolBalances, "unready", hbciBalances.getUnreadyBalance());
+                add(protocolBalances, "used", hbciBalances.getUsedBalance());
+            }
+            return protocolBalances;
+        }
+
+        static void add(List<Balance> protocolBalances, String type, de.adorsys.multibanking.domain.Balance hbciBalance) {
+            if (hbciBalance != null) {
```
Comment (valb3r) [inline opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java:99]: Return immediately if `hbciBalance` is `null`

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java [hunk 3]
```diff
@@ -70,7 +78,34 @@ public PaymentInfoBody extractPaymentInfoBody(ProcessResponse result) {
...
+                protocolBalances.add(Balance.builder()
+                    .balanceType(type)
+                    .lastChangeDateTime(hbciBalance.getDate().atStartOfDay(ZoneId.of("Europe/Berlin")).toOffsetDateTime())
```
Comment (valb3r) [inline opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java:102]: We should avoid using non UTC date/times at least on our side of responsibility. Please use ZoneId.UTC

**Review by valb3r — COMMENTED** [review]
Some comments tp fix. Please create a ticket to create Wiremock tests for these changes

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java (hunk 1 above)
Comment (electronicpeter) [reply opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java:88]: done

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java (hunk 2 above)
Comment (electronicpeter) [reply opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java:99]: done

File: opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java (hunk 3 above)
Comment (electronicpeter) [reply opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/entrypoint/HbciResultBodyExtractor.java:102]: done

**Review by valb3r — APPROVED** [review]

**Pull request merged**